File: app-website/trade-it/src/app/models/productData.ts
import { User } from "./user";

export interface ProductData {
    productId: number;
    title: string;
    descrioption: string;
    city: string;
    area: string;
    status: string;
    category: string;
    subcategory: string;
    postDateTime: string;
    price: number;
    requestCount: number | null;
    viewsCount: number | null;
    owner: User;
    instructions: string[];
    productImages: string[];
}

File: Backend (ASP .Net Web Api)/API/API Layer/DataModels/ProductData.cs
using DatabaseLayer.Models;

namespace API_Layer.DataModels
{
    public class ProductData
    {
        public int ProductId { get; set; }
        public string Title { get; set; }
        public string Descrioption { get; set; }
        public string City { get; set; }
        public string Area { get; set; }
        public string Status { get; set; }
        public string Category { get; set; }
        public string Subcategory { get; set; }
        public DateTime PostDateTime { get; set; }
        public double Price { get; set; }
        public int? RequestCount { get; set; }
        public int? ViewsCount { get; set; }
        //public User Owner { get; set; }
        public string OwnerFullName { get; set; }
        public string OwnerPhoneNumber { get; set; }
        public List<string> Instructions { get; set; }
        public List<string> ProductImages { get; set; }

    }
}
